tools/class-redirector/class-redirector/ClassRedirectorTask.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using System.Xml.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+#nullable enable
+
+namespace ClassRedirector
+{
+	public class ClassRedirectorTask : Microsoft.Build.Utilities.Task
+	{
+		public override bool Execute ()
+		{
+			if (!Directory.Exists (InputDirectory)) {
+				Log.LogError ($"InputDirectory {InputDirectory} doesn't exist.");
+				return false;
+			}
+
+			if (!DirectoryIsWritable (InputDirectory)) {
+				Log.LogError ($"InputDirectory {InputDirectory} is not writable.");
+				return false;
+			}
+
+			if (!File.Exists (ClassMapPath)) {
+				Log.LogError ($"ClassMapPath file {ClassMapPath} does not exist.");
+				return false;
+			}
+
+			var dllsToProcess = CollectDlls (InputDirectory);
+			var xamarinDll = FindXamarinDll (dllsToProcess);
+
+			if (xamarinDll is null) {
+				Log.LogError ($"unable to find platform dll in {InputDirectory}");
+				return false;
+			}
+
+			var map = ReadRegistrarFile (ClassMapPath);
+
+			try {
+				Log.LogMessage ($"Redirecting class_handle usage from directory {InputDirectory} in the following dlls: {string.Join (",", dllsToProcess)}");
+				Log.LogMessage ($"Redirecting class_handle usage with the platform dll {xamarinDll}");
+				Log.LogMessage ($"Redirecting class_handle usage with the following {nameof (ClassMapPath)}: {ClassMapPath}");
+				var rewriter = new Rewriter (map, xamarinDll, dllsToProcess);
+				rewriter.Process ();
+			} catch (Exception e) {
+				Log.LogErrorFromException (e);
+			}
+
+			return true;
+		}
+
+		[Required]
+		public string InputDirectory { get; set; } = "";
```

Comment:
This avoids a string allocation:

```suggestion
		public string InputDirectory { get; set; } = string.Empty;
```

Nit: we typically put the task input properties at the top of the class.